src/RadixDlt.NetworkGateway.GatewayApi/gateway-api-schema.yaml

```diff
@@ -1738,74 +1738,86 @@ components:
               $ref: "#/components/schemas/ResourceAddress"
 
     #
-    # /entity/* Request & Response Types
+    # /state/non-fungible/* Request & Response Types
     #
 
-    NonFungibleIdsRequest:
+    StateNonFungibleIdsRequest:
       allOf:
         - $ref: "#/components/schemas/AtLedgerStateMixin"
         - $ref: "#/components/schemas/CursorLimitMixin"
         - type: object
           required:
-            - address
+            - resource_address
           properties:
-            address:
-              $ref: "#/components/schemas/EntityAddress"
+            resource_address:
+              $ref: "#/components/schemas/ResourceAddress"
       example:
-        address: "<non-fungible-entity-address>"
-    NonFungibleIdsResponse:
+        resource_address: "<non-fungible-entity-address>"
+    StateNonFungibleIdsResponse:
       allOf:
         - $ref: "#/components/schemas/LedgerStateMixin"
         - type: object
           required:
-            - address
+            - resource_address
             - non_fungible_ids
           properties:
-            address:
-              $ref: "#/components/schemas/EntityAddress"
+            resource_address:
+              $ref: "#/components/schemas/ResourceAddress"
             non_fungible_ids:
               $ref: "#/components/schemas/NonFungibleIdsCollection"
 
-    NonFungibleDataRequest: # TODO rename to NonFungibleIdDataRequest
+    StateNonFungibleDetailsRequest:
```

Comment:
If we allow querying by multiple non_fungible_ids shouldn't it be named `StateNonFungible Ids DetailsRequest` ?

Reply:
Not really, not as far as I can tell. Actually here's some quick reading I've found on "nouns of nouns" - https://english.stackexchange.com/questions/255433/how-to-say-a-case-of-multiple-of-multiples-correctly